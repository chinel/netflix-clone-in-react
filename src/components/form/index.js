import React from "react";
import {
  Container,
  Base,
  Error,
  Title,
  Text,
  TextSmall,
  Link,
  Input,
  Submit,
} from "./styles/form";

export default function Form({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

Form.Error = function FormError({ children, ...restProps }) {
  return <Error {...restProps}>{children}</Error>;
};

Form.Base = function FormBase({ children, ...restProps }) {
  return <Base {...restProps}>{children}</Base>;
};
