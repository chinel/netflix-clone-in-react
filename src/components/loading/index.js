import React from "react";

export default function Loading({ src, ...restProps }) {
  return (
    <Spinner>
      <LockBody />
      <Picture src={`/images/users/${src}.png`}
    </Spinner>
  );
}
