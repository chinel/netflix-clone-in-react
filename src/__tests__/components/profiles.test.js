import React from "react";
import { render } from "@testing-library/react";
import { Profiles } from "../../components";

describe("<Profiles />", () => {
  it("renders the <Profiles /> with populated data", () => {
    const { container, getByText } = render(
      <Profiles>
        <Profiles.Title>Who's watching?</Profiles.Title>
        <Profiles.List>
          <Profiles.User onClick={() => {}}>
            <Profiles.Picture
              src="/images/nelly.png"
              data-testid="profile-picture"
            />
            <Profiles.Name>Nelly Kally</Profiles.Name>
          </Profiles.User>
        </Profiles.List>
      </Profiles>
    );
  });
});
