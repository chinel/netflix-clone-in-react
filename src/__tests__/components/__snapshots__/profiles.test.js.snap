// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profiles /> renders the <Profiles /> with populated data 1`] = `
<div
  class="profiles__Container-wp29q2-0 TeRKi"
>
  <h1
    class="profiles__Title-wp29q2-1 bBbvEg"
  >
    Who's watching?
  </h1>
  <ul
    class="profiles__List-wp29q2-2 dxUmQl"
  >
    <li
      class="profiles__Item-wp29q2-5 iCqkNr"
    >
      <img
        class="profiles__Picture-wp29q2-4 ggHnSq"
        data-testid="profile-picture"
        src="/images/users//images/nelly.png.png "
      />
      <p
        class="profiles__Name-wp29q2-3 lcuhlh"
      >
        Nelly Kally
      </p>
    </li>
  </ul>
</div>
`;
